osdep/win32/pthread.c

```diff
@@ -92,15 +91,38 @@ static int cond_wait(pthread_cond_t *restrict cond,
     return res ? 0 : ETIMEDOUT;
 }
 
+static struct timespec get_realtime(void)
+{
+    // same as mpv's code
+    union {
+        FILETIME ft;
+        ULARGE_INTEGER i;
+    } r;
+    GetSystemTimePreciseAsFileTime(&r.ft);
+    return (struct timespec) {
+        .tv_sec = r.i.QuadPart / UINT64_C(10000000),
+        .tv_nsec = r.i.QuadPart % UINT64_C(10000000)
+    };
+}
+
 int pthread_cond_timedwait(pthread_cond_t *restrict cond,
                            pthread_mutex_t *restrict mutex,
                            const struct timespec *restrict abstime)
 {
-    // mp time is not converted to realtime if internal pthread impl is used
-    int64_t now = mp_time_ns();
-    int64_t time_ns = abstime->tv_sec * UINT64_C(1000000000) + abstime->tv_nsec;
-    int64_t timeout_ms = (time_ns - now) / INT64_C(1000000);
-    return cond_wait(cond, mutex, MPCLAMP(timeout_ms, 0, INFINITE));
+    struct timespec ts = get_realtime();
+    DWORD timeout_ms = 0;
+    if (abstime->tv_sec >= INT64_MAX / 10000) {
```

Comment:
Should be `ULONG_MAX` for `DWORD` and `1000` for milliseconds

Reply:
The intent is to not have the `tv_sec * 1000LL` calculation overflow so I think it's right the way it is

Reply:
unless time_t is 32-bit. But ok. I mostly was wondering where `10000` comes from, `1000` should be enough? That said I'm not concerned about those sleeps that are longer than our lifetime, I will not see them fail :P